x/concentrated-liquidity/model/pool.go

```diff
@@ -238,11 +238,10 @@ func (p Pool) isCurrentTickInRange(lowerTick, upperTick int64) bool {
 
 // ApplySwap state of pool after swap.
 // It specifically overwrites the pool's liquidity, curr tick and the curr sqrt price
-func (p *Pool) ApplySwap(newLiquidity sdk.Dec, newCurrentTick sdk.Int, newCurrentSqrtPrice sdk.Dec) error {
+func (p *Pool) ApplySwap(newLiquidity sdk.Dec, newCurrentTick sdk.Int, newCurrentSqrtPrice sdk.Dec) {
```

Comment:
Should we add a validity check here or are we assuming the parameters are correctly passed in? e.g. return err if liq < 0 etc.

Reply:
Agree that it doesn't hurt to add validation but this is outside of the scope of this PR so I will make an issue

Reply:
https://github.com/osmosis-labs/osmosis/issues/4744